src/aeso_fcode_to_fate.erl

```diff
@@ -231,7 +231,28 @@ lookup_var(#env{vars = Vars}, X) ->
 
 %% -- The compiler --
 
+term_to_fate({lit, L}) ->
+     case L of
+        {int, N}             -> aeb_fate_data:make_integer(N);
+        {string, S}          -> aeb_fate_data:make_string(S);
+        {bool, B}            -> aeb_fate_data:make_boolean(B);
+        {account_pubkey, K}  -> aeb_fate_data:make_address(K);
+        {contract_pubkey, K} -> aeb_fate_data:make_contract(K);
+        {oracle_pubkey, K}   -> aeb_fate_data:make_oracle(K);
+        {oracle_query_id, _} -> ?TODO(fate_oracle_query_id_value)
+     end;
+term_to_fate(nil) ->
+    aeb_fate_data:make_list([]);
+term_to_fate({con, _Ar, I, As}) ->
+    %% what is Ar
+    aeb_fate_data:make_list([term_to_fate(I) | [ term_to_fate(A) || A<-As]]);
```

Comment:
`con` is not list cons, but variant type construction (`Ar` is the list of arities of the constructors, `I` is the tag index, and `As` is the constructor arguments). Cons is `{op, '::', [Hd, Tl]}`.